### DISCUSSION TITLE
Netspace hacked?

### DISCUSSION THREAD

**Discussion by fleance** [Chia-Network/chia-blockchain#5849] [General]
The netspace growth is starting to become suspicious.  By it's nature it should be a relatively linear slope, not asymptotic.  Growth is over 1EiB every 3 days roughly.  At an average 10 hr plot time that's 7.2 plot cycles in 3 days.  With 475k unique addresses that puts the avg number of plots per person, per cycle at 3 billion with rough math.

Even if you give 90% of all plotting to the top 10% that still comes out to the average user in the 90% is plotting 2,500 plots every 10 hours every day.

I believe that's at least 2 orders of magnitude off what can reasonably be expected and IMO that's a conservative estimate.  I haven't looked at NVME production numbers but I'm pretty sure those numbers exceed what is possible even if every high endurance NVME drive produced in 2020 were being used to plot Chia.  I believe it's about 11.5 billion years worth of plotting every 3 days @ 10hr avg.  Something doesn't add up.

**Comment by big-chia** [comment]
> With 475k unique addresses that puts the avg number of plots per person, per cycle at 3 billion with rough math.

Your math is way off here
1EiB = 1073700000 GiB
K32 plot is 101.4 GiB
~= 10,588,757 K32 plots
The mean average K32 plots per unique address for 1EiB of plots is therefore ~22.29212 or 2260.420968 GiB plots
As you state 1 EiB of growth is over approximately 3 days, that becomes ~7.43071 K32 plots or 753.473656 GiB plots per day
This amount of plots is achievable easily. I get ~1.2TiB per day.

When we consider that large farms will easily exceed the mean average it does not seem implausible that growth is so large.

Your point regarding NVME production seems to ignore production from previous years.

In my opinion, you are asking the wrong questions. The more interesting question to ask is where the huge amount of fiat currency used to purchase hundreds of millions USD worth of ssd and hard drives is coming from.
As of now, the current estimated network space is 9.143 EiB or 9,817,200,000 GiB. The average cost of hard disk storage is $0.02 per GiB, based on the cost of hard disks and cloud pricing (aws, etc.), therefore the estimated network space is using approximately $196,344,000 worth of hard disk storage.
Every 1EiB plots is using approximately $21,474,000 worth of hard disk storage.
The temp space requirements for a K32 on the wiki appears to be incorrect as all of mine are slightly larger, so I'll work with that figure
>Approximate working space used (without final file): 269.308 GiB

The temp space requirements are therefore ~2.66x the space required for the final plot
Of course the temp space can be reused, however if we go back to the 1EiB of growth in 3 days, and an average of 10 hours per plot, and 7.2 'cycles' in those 3 days, that would mean ~149,125,000 GiB of plots per cycle, and therefore 396,672,500 GiB of temp space.
This is equivalent to ~363,386 1.2TB (1091.6 GiB) ssd, e.g. Intel P3600 1.2TB ASP ~$150 = $54,507,900, however it is unlikely that number of old models are available, and such small ssd are probably unlikely to be used by large farmers. Large farmers are more likely to use larger ssds to make best usage of rack space.
396,672,500 GiB of temp space is equivalent to ~68,135 Intel P4618 6.4TB (5821.9 GiB) costs ~$2,800 = $190,778,000
Cost of temp space is awkward to calculate, however it is safe to assume there is somewhere between $50 million and $200 million worth of ssd temp space being used to plot the 1EiB growth per 3 days.

> Something doesn't add up.

Maybe. It's either groups with large financial backing, or a plot duplication bug

**Reply by Cactusbonus123** [reply]
Interesting data... The total EiB of manucactured ssd is 217 and this is owerall (only 1TB + ssd capacity can plotting effective)so  all of the manufactured ssd 0,5-1% plotting for right now? I don't think so....

**Reply by big-chia** [reply]
[StorageNewsletter](https://web.archive.org/web/20210215165825/https://www.storagenewsletter.com/2021/02/15/333-million-ssds-shipped-in-2020-corresponding-to-207eb/) state the total sales of SSD in 2020 was 207EB = 179.54388 EiB

396,672,500 GiB temp space is 0.2057% of the total sales of SSD in 2020
This figure is not completely infeasible, again considering not all SSDs in use will have been manufactured in 2020
Even accounting for only 1TB+ SSDs, which from [this source](https://www.gminsights.com/industry-analysis/solid-state-drive-ssd-market) in [this graph](https://www.gminsights.com/assets/img/uk-solid-state-drive-market-by-storage.png) and assuming UK market size is a rough approximation of the different capacities manufactured it would seem that roughly 50% of SSDs manufactured are 1TB or more

I do not question the capacities. The increase is large, yes, but not unrealistic given the appropriate financial backing. I would question the source of funds behind the large farms, and in turn what the money earned from block rewards will be used for. 

From [hpool's statistics](https://hpool.com/statistics/chia) it is interesting to note that the top user has 88.87 PB, close to 2 million USD in hard drives. The total of the top ten hpool users is nearly 9 million USD in hard drives.
Now hpool is only 40% of the netspace, and their block rewards are actually closer to 38% of total.

There is a very large amount of money being thrown at chia, and millions in fiat value mined every day.

**Reply by fleance** [reply]
Looks like by this evening today will be 1 EiB added in 24 hours.  Tripple the production rate of 5/19-5/22.  The only legit way to increase plotting speed is to add hardware right?  So if it took 400,000 SSDs over 3 days, to reach that in 1 day would be 1.2 M SSD.  So 800,000 SSDs added in a day or two?  At $100 a pop $80M in SSD alone are being spend daily plus the $20M in HDD for the storage.

Even considering 100 18 TB drives per chassis that's what, 600+ $10k units being racked in a day or two and another $6M?

Is Chia really receiving over $100M a day in infrastructure hardware not including HVAC, DC real estate, labor, electric, etc...

I agree that there are some deep pockets investing right now but tripling growth at this scale in 3 days seems less likely than HPOOL discovering a method to plot 4x faster.

It took Google at least 10 years to reach 14 EiB in infrastructure and Chia farmers are on pace to do that in a week?

**Reply by CCarroz** [reply]
The one issue with this argument is, your figures is assuming the cost is at list.  A bulk purchase could easily knock  50% off the price of list, depending on what is being purchased and how much.  The other thing is you have IT recycling shops that get a mass amount of old HDDs and SSDs. My friend owns such a shop and I could walk in there and pick up 1000 1TB(he said he could let a 1TB drive go for around $15) drives without blinking a eye. However 4TB drives is about the biggest drives he has and almost all of the 4TB drives are tied to old NetApp FAS and thus those drives can't be used in another system.  All I am trying to get at is, if you have a place to put the HW, and the power and cooling to run it, it doesn't seem too unfeasible to me.

**Comment by greyslater** [comment]
How is the total network space monitored btw if even my own farmer doesn't know the total capacity from all of my harvesters?

**Reply by fleance** [reply]
I am looking at the chia blockchain explorer linked to from the main page or client myself.

**Reply by greyslater** [reply]
it's a little bit off-topic here because it has nothing to do with the netspace hacked or not

but the total netspace showed to us is based on the diff and won blocks, and could be a bit higher, the netdiff does not adjust instantly
That's insane

**Comment by chiabug** [comment]
The biggest farms likely are not just spending money on hardware.  It is likely that they have also modified the code to incorporate efficiencies that are currently not in the general release.  Such mods might give them a 20 or 30 percent plotting advantage.  Regardless they can't increase the number of blocks available to win so they can only hope for a bigger share of the blocks that are "found".  An interesting stat would be an accurate number of the total blocks that are actually being won by farms with less than 30 TiB.  I am not sure such a number can actually be computed given that large farms can simply have many addresses.
